### PR TITLE
89 queue buffer overflow

### PR DESCRIPTION
**Pull request by manuelcueto**
relates to #89 
Publisher Actor now handles its internal queue's backpressuring when its buffer is full, thus avoiding the actor dying when it happens.

### CODE REVIEW

File: scheduler/src/main/scala/com/sky/kms/actors/PublisherActor.scala [hunk 1]
```diff
@@ -2,38 +2,54 @@ package com.sky.kms.actors
...
+import akka.actor.Stash
 
-class PublisherActor extends Actor with ActorLogging {
+class PublisherActor extends Actor with ActorLogging with Stash {
 
-  implicit val ec = context.dispatcher
-
-  override def receive: Receive = waitForQueue
+  implicit val ec                         = context.dispatcher
+  var stttt: Future[BufferUnblocked.type] = null
```
Comment (paulaylingdev) [inline scheduler/src/main/scala/com/sky/kms/actors/PublisherActor.scala]: 🤔 what is this?

File: scheduler/src/test/scala/com/sky/kms/unit/PublisherActorSpec.scala [hunk 2]
```diff
@@ -34,7 +43,27 @@ class PublisherActorSpec extends AkkaSpecBase with MockitoSugar {
       expectTerminated(publisherActor)
     }
 
+    "do not offer when queue backpressures" in new TestContext {
```
Comment (paulaylingdev) [inline scheduler/src/test/scala/com/sky/kms/unit/PublisherActorSpec.scala]: Maybe I am missing it but what is the assertion in this test?

**Comment by lacarvalho91** [comment]
this just moves the buffer into the actors *unbounded* mailbox

**Comment by lacarvalho91** [comment]
queue buffer overflow error generally just means the app is not configured appropriately to handle the load it is getting, with this change the buffer just becomes the actors mailbox so its the same really. Memory will just fill in the actors mailbox

**Pull request closed without merging**